src/appleseed/foundation/meta/tests/test_imagefilewriter.cpp

```diff
@@ -0,0 +1,245 @@
+        {
+            GenericImageFileReader reader;
+            std::unique_ptr<Image> image(reader.read(ImageFilePath));
+
+            for (size_t y = 0; y < 2; ++y)
+            {
+                for (size_t x = 0; x < 2; ++x)
+                {
+                    Color3b c;
+                    image->get_pixel(x, y, c);
+                    EXPECT_EQ(Color3b(50, 100, 150), c);
+                }
+            }
+        }
+    }
+    
+    TEST_CASE(WriteJPG_CorrectlyWritesImagePixels)
+    {
+        const char* ImageFilePath = "unit tests/outputs/test_imagefilewriter.jpg";
+        
+        {
+            Image image(2, 2, 2, 2, 3, PixelFormatFloat);
+            
+            image.clear(Color3f(0.6f, 0.7f, 0.8f));
+            
+            GenericImageFileWriter writer(ImageFilePath);
+            writer.append_image(&image);
+            writer.write();
+        }
+        
+        {
+            GenericImageFileReader reader;
+            std::unique_ptr<Image> image(reader.read(ImageFilePath));
+
+            for (size_t y = 0; y < 2; ++y)
+            {
+                for (size_t x = 0; x < 2; ++x)
+                {
+                    Color3f c;
+                    image->get_pixel(x, y, c);
+                    EXPECT_FEQ_EPS(Color3f(0.6f, 0.7f, 0.8f), c, 0.003f);
+                }
+            }
+        }
+    }
+    
+    TEST_CASE(WritePNG_CorrectlyWritesImagePixels)
+    {
+        const char* ImageFilePath = "unit tests/outputs/test_imagefilewriter.png";
+        
+        {
+            Image image(2, 2, 2, 2, 4, PixelFormatFloat);
+            Color4b color(50, 100, 150, 42);
+            color.unpremultiply_in_place();
+            color.rgb() = fast_linear_rgb_to_srgb(color.rgb());
+            color = saturate(color);
+            color.premultiply_in_place();
+            
+            image.clear(color);
+            
+            GenericImageFileWriter writer(ImageFilePath);
+            writer.append_image(&image);
+            writer.write();
+        }
+        
+        {
+            GenericImageFileReader reader;
+            std::unique_ptr<Image> image(reader.read(ImageFilePath));
+            
+            Color4b color(50, 100, 150, 42);
+            color.unpremultiply_in_place();
+            color.rgb() = fast_linear_rgb_to_srgb(color.rgb());
+            color = saturate(color);
+            color.premultiply_in_place();
+            
+            for (size_t y = 0; y < 2; ++y)
+            {
+                for (size_t x = 0; x < 2; ++x)
+                {
+                    Color4b c;
+                    image->get_pixel(x, y, c);
+                    
+                    EXPECT_EQ(color, c);
+                }
+            }
+        }
+    }
+    
+    TEST_CASE(WriteTIFF_CorrectlyWritesImagePixels)
+    {
+        const char* ImageFilePath = "unit tests/outputs/test_imagefilewriter.tiff";
+        
+        {
+            Image image(16, 16, 16, 16, 4, PixelFormatUInt8);
```

Comment:
Why you make a 16x16 image here but you use 2x2 everywhere else ?

Reply:
Hi, the reason is that an error was thrown when writing tiff files: "tile size have to be multiple of 16"

Reply:
A short comment explaining this would be useful.